urdf/CMakeLists.txt

```diff
@@ -17,53 +14,61 @@ endif()
+set(generated_install_path "${CMAKE_CURRENT_BINARY_DIR}/include")
+set(generated_compat_header "${generated_install_path}/${PROJECT_NAME}/urdfdom_compatibility.h")
+include_directories(${generated_install_path})
 configure_file(urdfdom_compatibility.h.in "${generated_compat_header}" @ONLY)
 
-add_compile_options(-std=c++11)
+# Default to C++11
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 11)
+endif()
 
-catkin_package(
-  LIBRARIES ${PROJECT_NAME}
-  INCLUDE_DIRS include ${CATKIN_DEVEL_PREFIX}/include
-  CATKIN_DEPENDS rosconsole_bridge roscpp
-  DEPENDS urdfdom_headers urdfdom Boost TinyXML
-)
-install(FILES ${generated_compat_header} DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION})
+install(FILES ${generated_compat_header} DESTINATION include/${PROJECT_NAME})
 
 include_directories(
   include
-  ${Boost_INCLUDE_DIR}
-  ${catkin_INCLUDE_DIRS}
   ${urdfdom_INCLUDE_DIRS}
   ${urdfdom_headers_INCLUDE_DIRS}
-  ${TinyXML_INCLUDE_DIRS}
-  )
+  ${TinyXML_INCLUDE_DIRS})
 
-link_directories(${Boost_LIBRARY_DIRS} ${catkin_LIBRARY_DIRS})
+add_library(${PROJECT_NAME} SHARED src/model.cpp)
+target_link_libraries(${PROJECT_NAME} ${TinyXML_LIBRARIES} ${urdfdom_LIBRARIES})
 
-add_library(${PROJECT_NAME} src/model.cpp src/rosconsole_bridge.cpp)
-target_link_libraries(${PROJECT_NAME} ${TinyXML_LIBRARIES} ${catkin_LIBRARIES} ${urdfdom_LIBRARIES})
+if(WIN32)
+  target_compile_definitions(${PROJECT_NAME} PRIVATE "URDF_BUILDING_DLL")
+endif()
 
 if(APPLE)
   set_target_properties(${PROJECT_NAME} PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
-endif(APPLE)
-
-if(CATKIN_ENABLE_TESTING)
-  find_package(catkin REQUIRED COMPONENTS rostest)
-  add_rostest_gtest(test_urdf_parser test/test_robot_model_parser.launch test/test_robot_model_parser.cpp)
-  target_link_libraries(test_urdf_parser ${PROJECT_NAME})
-
-  catkin_add_gtest(urdfdom_compatibility_test test/urdfdom_compatibility.cpp)
 endif()
 
-# no idea how CATKIN does this
-# rosbuild_add_rostest(${PROJECT_SOURCE_DIR}/test/test_robot_model_parser.launch)
-
-install(TARGETS ${PROJECT_NAME} 
-  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION})
+install(TARGETS ${PROJECT_NAME}
+  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib
+  RUNTIME DESTINATION bin)
 
 install(DIRECTORY include/${PROJECT_NAME}/
-  DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION})
+  DESTINATION include/${PROJECT_NAME})
+
+if(BUILD_TESTING)
+  find_package(ament_cmake_cppcheck REQUIRED)
+  find_package(ament_cmake_cpplint REQUIRED)
+  find_package(ament_cmake_lint_cmake REQUIRED)
+  find_package(ament_cmake_uncrustify REQUIRED)
+  # This forces cppcheck to consider all files in this project to be C++,
+  # including the headers which end with .h, which cppcheck would normally
+  # consider to be C instead.
+  ament_cppcheck(LANGUAGE "c++")
+  ament_cpplint()
+  ament_lint_cmake()
+  ament_uncrustify()
```

Comment:
use ament lint auto rather than calling out each linter? (maybe that prevents from passing the LANGUAGE to cppcheck?)

Reply:
I'll give it a try.  It may involve more style changes.

Reply:
If it prevents you from passing the option you want to cppcheck, let's not bother about it because that'll result in C style changes for C++ code.

Reply:
Ah, I see.  So `ament_lint_auto` is currently passing for everything except for copyright notices.  Since we don't have https://github.com/ament/ament_lint/issues/75 fixed yet, it is getting tripped up on some small differences in how the BSD license in some of the files is laid out.  However, I can disable *just* the copyright tests, and things seem to pass, so I'll go with that for now.

Reply:
cool:

> I can disable just the copyright tests, and things seem to pass, so I'll go with that for now.

you can do this by using:
```
find_package(ament_lint_auto REQUIRED)  
set(ament_cmake_copyright_FOUND TRUE)
ament_lint_auto_find_test_dependencies()
```